### PR TITLE
Properly normalize paths for event, no &$magic needed

### PR DESCRIPTION
**Pull request by butonic**
The paths in the begin and after events are currently not normalized, leading to double slashes in logged paths.

This is a safeguard against broken code as in https://github.com/owncloud/core/pull/31406

### CODE REVIEW

File: lib/private/Files/View.php [hunk 1]
```diff
@@ -891,20 +891,21 @@ public function rename($path1, $path2) {
...
-				and !Filesystem::isForbiddenFileOrDir($path2)
+				Filesystem::isValidPath($absolutePath2)
+				&& Filesystem::isValidPath($absolutePath1)
+				&& !Filesystem::isForbiddenFileOrDir($absolutePath2)
 			) {
 				$path1 = $this->getRelativePath($absolutePath1);
 				$path2 = $this->getRelativePath($absolutePath2);
 
-				if ($path1 == null or $path2 == null) {
+				if ($path1 === null || $path2 === null) {
```
Comment (tomneedham) [inline lib/private/Files/View.php]: * shudders *

Comment (butonic) [reply lib/private/Files/View.php]: is that a +1?

File: tests/lib/Files/ViewTest.php [hunk 2]
```diff
@@ -462,12 +462,16 @@ public function copyBetweenStorages($storage1, $storage2) {
...
-		$rootView->copy('substorage', 'anotherfolder');
+		// paths should not start or end in /
+		$rootView->copy('substorage/', '/anotherfolder');
 		$this->assertArrayHasKey('path', $calledEvent[1]);
 		$this->assertEquals('file.aftercreate', $calledEvent[0]);
 		$this->assertArrayHasKey('oldpath', $calledCopyEvent[1]);
 		$this->assertArrayHasKey('newpath', $calledCopyEvent[1]);
+		// mak sure paths in events have been normalized
 		$this->assertEquals('file.aftercopy', $calledCopyEvent[0]);
+		$this->assertSame('/substorage', $calledCopyEvent[1]['oldpath']);
```
Comment (butonic) [inline tests/lib/Files/ViewTest.php]: @PVince81 I now deliberately use paths starting with a `/`. without this PR this fails. There might be other events that do not normalize the path before emitting them in event, but should be checked separately.

**Comment by butonic** [comment]
@PVince81 @sharidas ready to review

**Review by PVince81 — APPROVED** [review]
:+1: looks good

**Comment by phil-davis** [comment]
Backport ``stable10`` #31689

**Pull request merged**